jupyterhub/spawner.py

```diff
@@ -56,6 +56,25 @@ class Spawner(LoggingConfigurable):
     orm_spawner = Any()
     user = Any()
 
+    def __init_subclass__(cls, **kwargs):
+        super().__init_subclass__()
+
+        missing = []
+        not_coro = []
+        for attr in ('start','stop', 'poll'):
+            if getattr(Spawner,attr) is getattr(cls, attr):
+                missing.append(attr)
+            if not getattr(getattr(cls, attr), '__tornado_coroutine__', None):
```

Comment:
Can we remove the coroutine check? I don't think it solves a problem and it introduces an artificially strict check. For instance, a method that returns a Future is perfectly valid. All that really matters is that the return value is awaitable.